### PR TITLE
Avoid "must not be null" gql errors

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/gqlgen/request.go [hunk 1]
```diff
@@ -74,12 +74,12 @@ func (r *requestResolver) PotentialProviders(ctx context.Context, obj *models.Po
...
 func (r *requestResolver) Organization(ctx context.Context, obj *models.Post) (*models.Organization, error) {
 	if obj == nil {
-		return nil, nil
+		return &models.Organization{}, nil
 	}
 
 	organization, err := obj.GetOrganization()
 	if err != nil {
-		return nil, domain.ReportError(ctx, err, "GetRequestOrganization")
+		return &models.Organization{}, domain.ReportError(ctx, err, "GetRequestOrganization")
```
Comment (Schparky) [inline application/gqlgen/request.go]: `organization` is nullable too

File: application/gqlgen/request.go [hunk 2]
```diff
@@ -128,12 +128,12 @@ func (r *requestResolver) Destination(ctx context.Context, obj *models.Post) (*m
...
 	if obj == nil {
-		return nil, nil
+		return &models.Location{}, nil
 	}
 
 	origin, err := obj.GetOrigin()
 	if err != nil {
-		return nil, domain.ReportError(ctx, err, "GetRequestOrigin")
+		return &models.Location{}, domain.ReportError(ctx, err, "GetRequestOrigin")
 	}
```
Comment (Schparky) [inline application/gqlgen/request.go]: `origin` is nullable

File: application/gqlgen/request.go [hunk 3]
```diff
@@ -180,12 +180,12 @@ func (r *requestResolver) Kilograms(ctx context.Context, obj *models.Post) (*flo
...
 func (r *requestResolver) Photo(ctx context.Context, obj *models.Post) (*models.File, error) {
 	if obj == nil {
-		return nil, nil
+		return &models.File{}, nil
 	}
 
 	photo, err := obj.GetPhoto()
 	if err != nil {
-		return nil, domain.ReportError(ctx, err, "GetRequestPhoto")
+		return &models.File{}, domain.ReportError(ctx, err, "GetRequestPhoto")
```
Comment (Schparky) [inline application/gqlgen/request.go]: `photo` is nullable

File: application/gqlgen/request.go [hunk 4]
```diff
@@ -225,12 +225,12 @@ func (r *requestResolver) Files(ctx context.Context, obj *models.Post) ([]models
...
 func (r *requestResolver) Meeting(ctx context.Context, obj *models.Post) (*models.Meeting, error) {
 	if obj == nil {
-		return nil, nil
+		return &models.Meeting{}, nil
 	}
 
 	meeting, err := obj.Meeting()
 	if err != nil {
-		return nil, domain.ReportError(ctx, err, "GetRequestMeeting")
+		return &models.Meeting{}, domain.ReportError(ctx, err, "GetRequestMeeting")
```
Comment (Schparky) [inline application/gqlgen/request.go]: `meeting` is nullable

File: application/gqlgen/user.go [hunk 5]
```diff
@@ -96,12 +96,12 @@ func (r *userResolver) PhotoID(ctx context.Context, obj *models.User) (*string,
...
 func (r *userResolver) Location(ctx context.Context, obj *models.User) (*models.Location, error) {
 	if obj == nil {
-		return nil, nil
+		return &models.Location{}, nil
 	}
 
 	location, err := obj.GetLocation()
 	if err != nil {
-		return nil, domain.ReportError(ctx, err, "GetUserLocation")
+		return &models.Location{}, domain.ReportError(ctx, err, "GetUserLocation")
```
Comment (Schparky) [inline application/gqlgen/user.go]: `location` is nullable

File: application/gqlgen/user.go [hunk 6]
```diff
@@ -175,12 +175,12 @@ func (r *queryResolver) User(ctx context.Context, id *string) (*models.User, err
...
 			"role": role,
 		}
-		return nil, domain.ReportError(ctx, err, "GetUser.Unauthorized", extras)
+		return &models.User{}, domain.ReportError(ctx, err, "GetUser.Unauthorized", extras)
 	}
 
 	dbUser := models.User{}
 	if err := dbUser.FindByUUID(*id); err != nil {
-		return nil, domain.ReportError(ctx, err, "GetUser")
+		return &models.User{}, domain.ReportError(ctx, err, "GetUser")
```
Comment (Schparky) [inline application/gqlgen/user.go]: not sure if it should be, but the `user` response nullable

File: application/gqlgen/user_test.go [hunk 7]
```diff
@@ -27,39 +27,32 @@ func (gs *GqlgenSuite) Test_getPublicProfile() {
...
 
-			if test.wantNil {
-				gs.Nil(profile)
-				return
-			}
 			gs.NotNil(profile)
-			gs.Equal(test.want.ID, profile.ID, "ID doesn't match")
-			gs.Equal(test.want.Nickname, profile.Nickname, "Nickname doesn't match")
-			gs.Equal(*test.want.AvatarURL, *profile.AvatarURL, "AvatarURL doesn't match")
+			gs.Equal(test.want, profile, "incorrect provfile")
```
Comment (Schparky) [inline application/gqlgen/user_test.go]:
```suggestion
			gs.Equal(test.want, profile, "incorrect profile")
```

File: application/gqlgen/meeting.go [hunk 8]
```diff
@@ -89,12 +89,12 @@ func (r *meetingResolver) EndDate(ctx context.Context, obj *models.Meeting) (str
 // ImageFile retrieves the file associated with the meeting
 func (r *meetingResolver) ImageFile(ctx context.Context, obj *models.Meeting) (*models.File, error) {
 	if obj == nil {
-		return nil, nil
+		return &models.File{}, nil
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/meeting.go]: Oops. Silly me. I didn't consider how it was related to whether something is nullable in the schema file.  Good catch.

**Pull request merged**